### PR TITLE
Move ownership of subscriptions to `ConnectionHandler`

### PR DESCRIPTION
**Pull request by caspervonb**

### CODE REVIEW

File: async-nats/src/lib.rs [hunk 1]
```diff
@@ -223,12 +223,13 @@ pub enum Command {
...
     Subscribe {
-        sid: u64,
+        id: u64,
         subject: String,
         queue_group: Option<String>,
+        sender: mpsc::Sender<Message>,
     },
     Unsubscribe {
-        uid: u64,
+        id: u64,
```
Comment (Jarema) [inline async-nats/src/lib.rs:232]: `id` is effectivly `sid`, right? No reason to not call it `sid`.

Comment (caspervonb) [reply async-nats/src/lib.rs:232]: Did not re-implement remapping here, so yes.

Comment (Jarema) [reply async-nats/src/lib.rs:232]: so lets just call it sid :)

**Review by Jarema — APPROVED** [review]
LGTM

**Pull request merged**